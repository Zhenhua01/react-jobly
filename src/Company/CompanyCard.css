.logo{
    width: 50px;
    height: auto;
    /* position: relative; */
    /* right: 0; */
    float: right;
    margin-left: 3rem;
}

h6{
    margin-top: 0;
}

.companyCard{
    color: black;
    /* background-color: white; */
    position: relative;
    display:block;
    box-sizing: border-box;
    padding: 1rem;
    text-align:left;
    /* border: 1px black solid; */
}

a{
    color: black;
}