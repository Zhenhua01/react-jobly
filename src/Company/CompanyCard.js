import { Link } from 'react-router-dom';

/**
 * Presentational Component for displaying each company
 *
 * Props: company: { handle, name, description }
 *
 * RoutesList -> CompanyList -> {SearchForm, CompanyCard}
 */
function CompanyCard({ company }) {
  const { handle, name, description, logoUrl } = company;
  
  return (
    <Link to={`/companies/${handle}`} key={handle}>
      <div className='CompanyCard'>
        <b>{name}</b>
        <p>{description}</p>
        { logoUrl &&
        <img src={logoUrl} alt={`${name}-logo`} />}
      </div>
    </Link>
  );

};

export default CompanyCard;