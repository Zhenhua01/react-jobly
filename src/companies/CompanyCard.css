.link {
    text-decoration: none;
}

.logo {
    width: 50px;
}

.CompanyCard {
    color: black;
    box-sizing: border-box;
    padding: 1rem;
    text-align: left;
}

.CompanyCard:hover {
    color: gray
}