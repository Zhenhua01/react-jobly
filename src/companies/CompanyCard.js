import { Link } from 'react-router-dom';
import './CompanyCard.css';


/** CompanyCard component for displaying information of each company.
 *
 * Props: company: { handle, name, description, logoUrl }
 *
 * Context: user
 *  { username, firstName, lastName, email, isAdmin, applications:[] }
 *
 * RoutesList -> CompanyList -> { CompanyCard, SearchForm }
 */

function CompanyCard({ company }) {
  const { handle, name, description, logoUrl } = company;

  return (
    <div className='container-fluid col-8 bg-white rounded'>
      <Link className="link" to={`/companies/${handle}`} key={handle}>
        <div className='CompanyCard mb-4'>

          <h5><u>{ name }</u>
            {logoUrl &&
              <img className='logo float-end' src={logoUrl} alt={`${name}-logo`} />}
          </h5>
          <p>{description}</p>

        </div>
      </Link>
    </div>
  );

};

export default CompanyCard;