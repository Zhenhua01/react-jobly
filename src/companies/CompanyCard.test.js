import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

const company1 = {
  handle: "rithm",
  name: "Rithm School",
  description: "Become an exceptional developer in 16 weeks.",
  logo_url: "https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
};

const company2 = {
  handle: "algo",
  name: "Algo School",
  description: "Become a mediocre developer in 160 weeks."
};

it("matches snapshot with logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={company1} />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={company2}/>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
