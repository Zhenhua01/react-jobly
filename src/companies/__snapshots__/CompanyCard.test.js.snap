// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <div
    class="container m-2 companyCard shadow p-3 mb-3 bg-white rounded"
  >
    <a
      href="/companies/rithm"
    >
      <div
        class="CompanyCard"
      >
        <h6>
          Rithm School
        </h6>
        <p>
          Become an exceptional developer in 16 weeks.
        </p>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <div
    class="container m-2 companyCard shadow p-3 mb-3 bg-white rounded"
  >
    <a
      href="/companies/algo"
    >
      <div
        class="CompanyCard"
      >
        <h6>
          Algo School
        </h6>
        <p>
          Become a mediocre developer in 160 weeks.
        </p>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
